src/tests/end-to-end/modeler/test-modeler.cpp

```diff
@@ -0,0 +1,139 @@
+// Copyright 2007-2025, RTE (https://www.rte-france.com)
+// See AUTHORS.txt
+// SPDX-License-Identifier: MPL-2.0
+// This file is part of Antares-Simulator,
+// Adequacy and Performance assessment for interconnected energy networks.
+//
+// Antares_Simulator is free software: you can redistribute it and/or modify
+// it under the terms of the Mozilla Public Licence 2.0 as published by
+// the Mozilla Foundation, either version 2 of the License, or
+// (at your option) any later version.
+//
+// Antares_Simulator is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// Mozilla Public Licence 2.0 for more details.
+//
+// You should have received a copy of the Mozilla Public Licence 2.0
+// along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+
+#define BOOST_TEST_MODULE testE2EModeler
+#include <boost/test/unit_test.hpp>
+
+#include <antares/solver/modeler/ILoader.h>
+#include <antares/solver/modeler/Modeler.h>
+#include "antares/expressions/nodes/GreaterThanOrEqualNode.h"
+#include "antares/solver/modeler/IWriter.h"
+
+#include "inmemory-modeler.h"
+
+class EmptyDataSeries: public Antares::Optimisation::LinearProblemApi::ILinearProblemData
+{
+public:
+    double getData([[maybe_unused]] const std::string& dataSetId,
+                   [[maybe_unused]] const std::string& scenarioGroup,
+                   [[maybe_unused]] unsigned year,
+                   [[maybe_unused]] unsigned hour) override
+    {
+        return 0.;
+    }
+};
+
+Antares::ModelerStudy::SystemModel::Component copyComponent(
+  const Antares::ModelerStudy::SystemModel::Component& c)
+{
+    Antares::ModelerStudy::SystemModel::ComponentBuilder builder;
+    return builder.withId(c.Id())
+      .withModel(c.getModel())
+      .withScenarioGroupId(c.getScenarioGroupId())
+      .withParameterValues(c.getParameterValues())
+      .build();
+}
+
+using Models = std::unordered_map<std::string, Antares::ModelerStudy::SystemModel::Model>;
+
+template<class Fixture>
+class InMemoryLoader: public Antares::Solver::ILoader
+{
```

Comment:
This template is instanciated only with one type, considering removing template to make code easier to read

Reply:
One type so far.
For further test we may provide a different fixture.